modules/database/funcs.go

```diff
@@ -262,9 +258,44 @@ func (d *Database) WaitForDBCreated(
 	return d.WaitForDBCreatedWithTimeout(ctx, h, time.Second*5)
 }
 
-//
+// Get MariaDB instance
+func (d *Database) getDBInstance(
+	ctx context.Context,
+	h *helper.Helper,
+) error {
+	namespace := d.namespace
+	if namespace == "" {
+		namespace = h.GetBeforeObject().GetNamespace()
+	}
+	// Fetch the MariaDB instance
+	dbInstance := &mariadbv1.MariaDB{}
+	err := h.GetClient().Get(
+		ctx,
+		types.NamespacedName{
+			Namespace: namespace,
+			Name:      "openstack",
+		},
+		dbInstance)
+	if err != nil {
+		if k8s_errors.IsNotFound(err) {
+			return util.WrapErrorForObject(
+				fmt.Sprintf("Failed to get db instance in %s", namespace),
```

Comment:
I would call this MariaDB instance as `db` is pretty overloaded term.

Reply:
AFAIK db instance vs db/database is not overloaded. We're not using the 'instance' in the programming sense here but as used with databases.

MariaaDB - db instance (database instance)
MariaDBDatabase - db, database